content/blog/2020-09-17-delivering-learnr-tutorials-in-a-package/index.Rmarkdown

```diff
@@ -0,0 +1,322 @@
+
+    ```{r, eval = FALSE}
+    devtools::install()
+    ```
+
+    Believe it or not, you've now made a package with a learnr tutorial inside!
+
+    You should test that your package has been properly created by calling the function that will load your tutorial from your installed package:
+
+1.  **Run your tutorial** with `learnr::run_tutorial("<tutorial-name>", "<package-name>")`.
+
+    ```{r, eval = FALSE}
+    learnr::run_tutorial("lesson1", "tutorialpackage")
+    ```
+
+    * The tutorial name here refers to the name of the `.Rmd` basename, not the user-facing title.
+    * The live tutorial will open up in a local browser window (*woot!*)
+    * This code is also how your users will be able to run your tutorial once they have installed your package.
+
+
+
+### (Optionally) Add dependencies for exercise checking
+
+If your tutorial includes [exercises that can be checked via an external package](https://rstudio.github.io/learnr/exercises.html#Exercise_Checking) (like [gradethis](https://rstudio-education.github.io/gradethis/), for example), you'll need to make sure this package gets installed along with your custom package. To do this, the external package needs to be added as an `Imports` dependency in the `DESCRIPTION` file. Once again, the usethis package will make this easier for us to do:
+
+
+1. **Run** `usethis::use_package("<external-package>")`
+
+    ```{r, eval = FALSE}
+    usethis::use_package("gradethis")
+    ```
+
+    * You should see something like this in the console output:
+
+    ```{r, eval = TRUE, echo = FALSE, out.width="70%"}
+    knitr::include_graphics("pkg-gradethis.png")
+    ```
+    
+2. You can confirm the package has been added as an import when you **open the `DESCRIPTION` file** in your project root.
+
+
+
+### (Optionally) Edit your package metadata
+
+This step is not necessary to have a working, usable package, but it creates a nicer experience for users interested in learning more about the contents of your package and how to use them.
+
+
+1. **Open the `DESCRIPTION` file** in your project root.
+
+
+2. **Edit the file** with your own information. Not sure what to write for each field? See [this resource](https://r-pkgs.org/description.html#pkg-description) for suggestions.
+
+3. Finally, make sure to __*Build* > *Install and Restart*__ your package again to include these changes.
+
+
+
+### Getting your tutorial to your audience
+
+Your tutorial package has (finally!) been built, now it's time to get it out there!
+
+1. **[Commit](https://happygitwithr.com/rstudio-git-github.html#make-local-changes-save-commit) and [push](https://happygitwithr.com/rstudio-git-github.html#push-your-local-changes-online-to-github)** to GitHub to share your tutorial package with the world.
+
+1. **Have users *install* your package** from GitHub with `remotes::install_github(“<your-repo>/<package-name>")`
```

Comment:
may want to mention that they will each HAVE to have the the learnr AND the remotes packages installed to run locally.

Reply:
Ah I see that below, I would group together.